### ISSUE TITLE
[Feature Request]: Openai_like embedding model integration

### ISSUE THREAD

**Issue by regybean** [run-llama/llama_index#14340]
### Feature Description

Openai_like LLMs are supported, however, it would be useful to do this with embedding models. This would allow users to integrate custom frameworks more easily. Namely, deepspeed fastgen model inference or vLLM, as currently I can't see an easy way to integrate local (or hosted) pipeline parallelised embedding models.

### Reason

There is no way to integrate a custom API endpoint with embedding models and only recognised frameworks are supported

### Value of Feature

This will increase compatibility and enable embedding models to be parallelised across GPUs with frameworks such as DeepSpeed-mii and vLLM (I assume).

**Comment by logan-markewich** [comment]
@regybean technically this already exists

`OpenAIEmbedding(model_name="...", api_key="fake", api_base="...")`

**Comment by whisper-bye** [comment]
+1

**Comment by Loongphy** [comment]
You can use DatabricksEmbedding. https://github.com/run-llama/llama_index/blob/044a439dc2fda53c991e49a43e4e8e652dd8a735/llama-index-integrations/embeddings/llama-index-embeddings-databricks/llama_index/embeddings/databricks/base.py#L22

**Comment by Tendo33** [comment]
same problem

**Comment by logan-markewich** [comment]
@Tendo33 @Loongphy @whisper-bye @regybean actually OpenAIEmbedding DOES support this

This code works I promise you

```python
OpenAIEmbedding(model_name="...", api_key="fake", api_base="...")
```

There's even a check for this in the code
https://github.com/run-llama/llama_index/blob/bf0c1744408bd57c6140606b37550edfb5d5f298/llama-index-integrations/embeddings/llama-index-embeddings-openai/llama_index/embeddings/openai/base.py#L304

**Comment by logan-markewich** [comment]
Of course, there could be a dedicated class for this, but the class already supports the above

**Comment by logan-markewich** [comment]
Here's a colab link with proof
https://colab.research.google.com/drive/1TwO8HsD_XuI9vxDfM_etN0yKqd-Lll4p?usp=sharing
